test/common/http/path_utility_test.cc

```diff
@@ -85,5 +85,24 @@ TEST_F(PathUtilityTest, NormalizeCasePath) {
 // "/../c\r\n\"  '\n' '\r' should be excluded by http parser
 // "/a/\0c",     '\0' should be excluded by http parser
 
+// Paths that are valid get normalized.
+TEST_F(PathUtilityTest, MergeSlashes) {
+  const std::vector<std::pair<std::string, std::string>> non_normal_pairs{
+      {"", ""},                        // empty
+      {"/a", "/a"},                    // no-op
+      {"//a/b/c", "/a/b/c"},           // double / start
+      {"/a//b/c", "/a/b/c"},           // double / in the middle
+      {"/a//b?a=///c", "/a/b?a=///c"}, // slashes in the query are ignored
+      {"/a//b?", "/a/b?"},             // empty query
```

Comment:
Should we test for paths with just fragments, and a combination of query params and fragments?